pkg/kvstore/allocator/allocator.go

```diff
@@ -700,67 +238,82 @@ func (a *Allocator) GetNoCache(ctx context.Context, key AllocatorKey) (idpool.ID
 
 // GetByID returns the key associated with an ID. Returns nil if no key is
 // associated with the ID.
-func (a *Allocator) GetByID(id idpool.ID) (AllocatorKey, error) {
-	if key := a.mainCache.getByID(id); key != nil {
-		return key, nil
-	}
-
-	v, err := kvstore.Get(path.Join(a.idPrefix, id.String()))
+func (k *kvstoreBackend) GetByID(id idpool.ID) (allocator.AllocatorKey, error) {
+	v, err := kvstore.Get(path.Join(k.idPrefix, id.String()))
 	if err != nil {
 		return nil, err
 	}
 
-	return a.keyType.PutKey(string(v))
+	return k.keyType.PutKey(string(v))
 }
 
-// Release releases the use of an ID associated with the provided key. After
-// the last user has released the ID, the key is removed in the KVstore and
-// the returned lastUse value is true.
-func (a *Allocator) Release(ctx context.Context, key AllocatorKey) (lastUse bool, err error) {
-	log.WithField(fieldKey, key).Info("Releasing key")
+// UpdateKey refreshes the record that this node is using this key -> id
+// mapping. When reliablyMissing is set it will also recreate missing master or
+// slave keys.
+func (k *kvstoreBackend) UpdateKey(ctx context.Context, id idpool.ID, key allocator.AllocatorKey, reliablyMissing bool) error {
```

Comment:
Nitpick - `reliablyMissing` isn't clear for me. What about `forceUpdate`? I see that this name was used before so feel free to ignore this comment.

Reply:
This was the name before, when this function was called `recreateMasterKey` in pkg/kvstore/allocator so I won't change it unless all the other reviewers also want it. It is a confusing name nonetheless.

Reply:
reliablyMissing is true when it is known that the key is missing and it's not an opportunistic update. This is the case when we re-create after having received a delete notification for a key that is owned by the local agent.